docs/admin/healthcheck.md

```diff
@@ -0,0 +1,248 @@
+**severe** impairment of functionality for users. To validate this
+functionality, Coder will periodically attempt to establish a WebSocket
+connection with itself using the configured [Access URL](#access-url), send a
+message over the connection, and attempt to read back that same message.
+
+### <a name="EWS01">EWS01: Failed to establish a WebSocket connection</a>
+
+**Problem:** Coder was unable to establish a WebSocket connection over its own
+Access URL.
+
+**Solution:** There are multiple possible causes of this problem:
+
+1. Ensure that Coder's configured Access URL can be reached from the server
+   running Coder, using standard troubleshooting tools like `curl`:
+
+   ```shell
+   curl -v "https://coder.company.com:443/"
+   ```
+
+2. Ensure that any reverse proxy that is sitting in front of Coder's configured
+   access URL is not stripping the HTTP header `Upgrade: websocket`.
+
+### <a name="EWS02">EWS02: Failed to echo a WebSocket message</a>
+
+**Problem:** Coder was able to establish a WebSocket connection, but was unable
+to write a message.
+
+**Solution:** There are multiple possible causes of this problem:
+
+1. Validate that any reverse proxy servers in front of Coder's configured access
+   URL are not prematurely closing the connection.
+2. Validate that the network link between Coder and the workspace proxy is
+   stable, e.g. by using `ping`.
+3. Validate that any internal network infrastructure (for example, firewalls,
+   proxies, VPNs) do not interfere with WebSocket connections.
+
+## Workspace Proxy
+
+If you have configured [Workspace Proxies](../admin/workspace-proxies.md), Coder
+will periodically query their availability and show their status here.
+
+### <a name="EWP01">EWP01: Error Updating Workspace Proxy Health</a>
+
+**Problem:** Coder was unable to query the connected workspace proxies for their
+health status.
+
+**Solution:** This may be a transient issue. If it persists, it could signify a
+connectivity issue.
+
+### <a name="EWP02">EWP02: Error Fetching Workspace Proxies</a>
+
+**Problem:** Coder was unable to fetch the stored workspace proxy health data
+from the database.
+
+**Solution:** This may be a transient issue. If it persists, it could signify an
+issue with Coder's configured database.
+
+### <a name="EWP03">EWP03: Workspace Proxy Version Mismatch</a>
+
+**Problem:** One or more workspace proxies are more than one major or minor
+version out of date with the main deployment. It is important that workspace
+proxies are updated at the same time as the main deployment to minimize the risk
+of API incompatibility.
+
+**Solution:** Update the workspace proxy to match the currently running version
+of Coder.
+
+### <a name="EWP04">EWP04: One or more Workspace Proxies Unhealthy</a>
+
+**Problem:** One or more workspace proxies are not reachable.
+
+**Solution:** Ensure that Coder can establish a connection to the configured
+workspace proxies on port 443.
```

Comment:
Ensure that Coder can establish a connection to the configured
workspace proxies

Reply:
:+1: